File: test/main/java/org/apache/prepbuddy/smoothers/SimpleSlidingWindowTest.java
package org.apache.prepbuddy.smoothers;

import org.apache.prepbuddy.SparkTestCase;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class SimpleSlidingWindowTest extends SparkTestCase {
    @Test
    public void shouldAddValueTillFullThanSwapTheValues() {
        SimpleSlidingWindow slidingWindow = new SimpleSlidingWindow(3);
        slidingWindow.add(3);
        slidingWindow.add(3);
        slidingWindow.add(3);

        assertEquals(3.0, slidingWindow.average(), 0.01);

        slidingWindow.add(9);
        assertEquals(5.0, slidingWindow.average(), 0.01);

        slidingWindow.add(15);
        assertEquals(9.0, slidingWindow.average(), 0.01);
    }
}

File: src/test/scala/org/apache/datacommons/prepbuddy/smoothers/SimpleSlidingWindowTest.scala
package org.apache.datacommons.prepbuddy.smoothers

import org.scalatest.FunSuite

class SimpleSlidingWindowTest extends FunSuite {
    test("should add value till window is full than move values to left") {
        val slidingWindow: SimpleSlidingWindow = new SimpleSlidingWindow(3)
        slidingWindow.add(3)
        slidingWindow.add(3)
        slidingWindow.add(3)

        assert(3.0 == slidingWindow.average, 0.01)

        slidingWindow.add(9)
        assert(5.0 == slidingWindow.average, 0.01)

        slidingWindow.add(15)
        assert(9.0 == slidingWindow.average, 0.01)
    }
}
